analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.RoslynCfg.cs

```diff
@@ -0,0 +1,111 @@
+ */
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.CFG.Roslyn;
+using SonarAnalyzer.Extensions;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    public sealed partial class MemberInitializerRedundant
+    {
+        private class RoslynChecker : CfgAllPathValidator
+        {
+            private readonly ISymbol memberToCheck;
+            private readonly ControlFlowGraph cfg;
+
+            public RoslynChecker(ControlFlowGraph cfg, ISymbol memberToCheck) : base(cfg)
+            {
+                this.cfg = cfg;
+                this.memberToCheck = memberToCheck;
+            }
+
+            // Returns true if the block contains assignment before access
+            protected override bool IsValid(BasicBlock block) =>
+                Match(block, checkReadBeforeWrite: false);
+
+            // Returns true if the block contains access before assignment
+            protected override bool IsInvalid(BasicBlock block) =>
+                Match(block, checkReadBeforeWrite: true);
+
+            private bool Match(BasicBlock block, bool checkReadBeforeWrite)
+            {
+                foreach (var operation in block.OperationsAndBranchValue)
+                {
+                    if (ProcessOperation(operation, cfg, checkReadBeforeWrite, out var result))
+                    {
+                        return result;
+                    }
+                }
+
+                return false;
+            }
+
+            private bool ProcessOperation(IOperation operation, ControlFlowGraph controlFlowGraph, bool isRead, out bool result)
+            {
+                foreach (var child in operation.DescendantsAndSelf().ToReversedExecutionOrder())
+                {
+                    if (isRead && child.Instance.Kind == OperationKindEx.FlowAnonymousFunction)
+                    {
+                        var anonymousFunctionCfg = controlFlowGraph.GetAnonymousFunctionControlFlowGraph(IFlowAnonymousFunctionOperationWrapper.FromOperation(child.Instance));
+                        foreach (var subOperation in anonymousFunctionCfg.Blocks.SelectMany(x => x.OperationsAndBranchValue).SelectMany(x => x.DescendantsAndSelf()))
+                        {
+                            if (ProcessOperation(subOperation, anonymousFunctionCfg, isRead, out result))
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                    else if (memberToCheck.Equals(MemberSymbol(child.Instance)))
+                    {
+                        result = IsReadOrWrite(child, isRead);
+                        return true;
+                    }
+                }
+
+                result = false;
+                return false;
+            }
+
+            private bool IsReadOrWrite(IOperationWrapperSonar child, bool isRead)
```

Comment:
To propagate the rename
```suggestion
            private bool IsReadOrWrite(IOperationWrapperSonar child, bool checkReadBeforeWrite)
```